### PR TITLE
Scheduler:  add example for onOptionChanged and edit currentDate, currentView, views.name, views.type

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello card](https://trello.com/c/1rkfenFG/1550-extend-topics-about-currentdate-and-currentview)

### CODE REVIEW

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentView.md [hunk 1]
```diff
@@ -10,7 +10,10 @@ firedEvents: optionChanged
 Specifies the currently displayed view. Accepts the view's [name](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views/name.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/#name') or [type](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views/type.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/#type').
 
 ---
-In all situations when more than one view meets the **currentView** condition, the first matching view from the [views](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/') array takes precedence.
+
+The **Scheduler** displays the first view from the [views](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/) array whose **name** or **type** matches the **currentView** value.
+
+To subscribe to changes of the current view, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) option.
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentView.md]:
```suggestion
To subscribe to changes of the current view, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) handler.
```

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md [hunk 2]
```diff
@@ -6,7 +6,12 @@ firedEvents: optionChanged
...
 ---
 ##### shortDescription
-Specifies a date displayed on the current scheduler [view](/concepts/05%20Widgets/Scheduler/060%20Views/010%20View%20Types '/Documentation/Guide/Widgets/Scheduler/Views/View_Types/') by default.
+Specifies the current date.
 
 ---
-This is the date which is displayed when rendering the widget. Although, an end-user can change the current date by using the Date Navigator.
\ No newline at end of file
+
+Users can change this value using the Date Navigator:
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md]: 1. I suggest you to add a link
2. using the Date Navigator -> in the `[Date Navigator](link)`

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md [hunk 3]
```diff
@@ -6,7 +6,12 @@ firedEvents: optionChanged
...
+
+![Scheduler Date Navigator](/images/UiWidgets/Scheduler_Date_Navigator.png)
+
+To subscribe to changes of the current date, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) option.
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md]:
```suggestion
To subscribe to changes of the current date, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) handler.
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 4]
```diff
@@ -24,3 +24,22 @@ The modified option if it belongs to the first level. Otherwise, the first-level
...
+---
+##### jQuery
+
+    <!--index.js-->
+    $(function() {
+        $("#{widgetName}Container").{WidgetName}({
+            // ...
+            onOptionChanged: function(e) {
+                if(e.name == "desiredOptionName") {
+                    // ...
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]: 1. I would use the name 'changedOption' and the strict equality
2. in the comment - // handle the option change here

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 5]
```diff
@@ -24,3 +24,22 @@ The modified option if it belongs to the first level. Otherwise, the first-level
 The modified option's new value.
 
 ---
+
+The following example shows how to subscribe to a desired option of {WidgetName}:
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]:
```suggestion
The following example shows how to subscribe to changes of an option of {WidgetName}:
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 6]
```diff
@@ -24,3 +24,122 @@ The modified option if it belongs to the first level. Otherwise, the first-level
 The modified option's new value.
 
 ---
+
+The following example shows how to subscribe to changes for an option of {WidgetName}:
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]:
```suggestion
The following example shows how to subscribe to changes of a **{WidgetName}** option:
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 7]
```diff
@@ -24,3 +24,122 @@ The modified option if it belongs to the first level. Otherwise, the first-level
...
+
+---
+
+##### jQuery
+
+    <!--index.js-->
+    $(function() {
+        $("#{widgetName}Container").{WidgetName}({
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]:
```suggestion
        $("#{widgetName}Container").dx{WidgetName}({
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 8]
```diff
@@ -24,3 +24,122 @@ The modified option if it belongs to the first level. Otherwise, the first-level
...
+    import Dx{WidgetName} from 'devextreme-vue/{widget-name}'; 
+
+    export default { 
+        components: { 
+            Dx{WidgetName}
+        }, 
+        // ...
+        methods: { 
+            onOptionChanged: function(args) {
+                if(args.name === "changedOption") {
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]:
```suggestion
            onOptionChanged: function(e) {
                if(e.name === "changedOption") {
```

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md [hunk 9]
```diff
@@ -6,7 +6,12 @@ firedEvents: optionChanged
...
 
 ---
-This is the date which is displayed when rendering the widget. Although, an end-user can change the current date by using the Date Navigator.
\ No newline at end of file
+
+Users can change this value in the [Date Navigator](/Documentation/Guide/Widgets/Scheduler/Date_Navigator/):
+
+![Scheduler Date Navigator](/images/UiWidgets/Scheduler_Date_Navigator.png)
+
+To subscribe to changes for the current date, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) handler.
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md]:
```suggestion
To subscribe to changes of the current date, use the [onOptionChanged](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#onOptionChanged) handler.
```

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md [hunk 10]
```diff
@@ -6,7 +6,12 @@ firedEvents: optionChanged
...
 ---
 ##### shortDescription
-Specifies a date displayed on the current scheduler [view](/concepts/05%20Widgets/Scheduler/060%20Views/010%20View%20Types '/Documentation/Guide/Widgets/Scheduler/Views/View_Types/') by default.
+Specifies the current date.
 
 ---
-This is the date which is displayed when rendering the widget. Although, an end-user can change the current date by using the Date Navigator.
\ No newline at end of file
+
+Users can change this value in the [Date Navigator](/Documentation/Guide/Widgets/Scheduler/Date_Navigator/):
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentDate.md]:
```suggestion
Users can change the current date in the [Date Navigator](/Documentation/Guide/Widgets/Scheduler/Date_Navigator/):
```

File: api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentView.md [hunk 11]
```diff
@@ -10,7 +10,10 @@ firedEvents: optionChanged
 Specifies the currently displayed view. Accepts the view's [name](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views/name.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/#name') or [type](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views/type.md '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/#type').
 
 ---
-In all situations when more than one view meets the **currentView** condition, the first matching view from the [views](/api-reference/10%20UI%20Widgets/dxScheduler/1%20Configuration/views '/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/') array takes precedence.
+
+The **Scheduler** displays the first view from the [views](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/) array whose **name** or **type** matches the **currentView** value.
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/dxScheduler/1 Configuration/currentView.md]: First, describe the situation in which this information matters.

```suggestion
When more than one view matches the **currentView** value, the **Scheduler** displays the first matching view from the [views](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/views/) array.
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md (hunk 6 above)
Comment (pavel-2020) [reply api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]: If WidgetName starts with a vowel (e.g. Action Sheet), the part of the sentence is "changes of **a A**ction Sheet option". So is "The following example shows how to subscribe to changes of an option" good?

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 12]
```diff
@@ -24,3 +24,121 @@ The modified option if it belongs to the first level. Otherwise, the first-level
...
 
 ---
+
+The following example shows how to subscribe to changes of a component option:
+
+---
+
+##### jQuery
+
+    $(function() {
```
Comment (RomanTsukanov) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md:35]:
```suggestion
    <!-- tab: index.js -->
    $(function() {
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 13]
```diff
@@ -24,3 +24,122 @@ The modified option if it belongs to the first level. Otherwise, the first-level
...
+    }) 
+
+    export class AppModule { }  
+
+##### Vue 
+  
+    <!-- tab: App.vue -->
+    <template> 
+        <Dx{WidgetName} ...
+            :on-option-changed="onOptionChanged">
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]: Vue requires the following syntax for handlers:
```suggestion
            @option-changed="onOptionChanged">            
```

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 14]
```diff
@@ -24,3 +24,122 @@ The modified option if it belongs to the first level. Otherwise, the first-level
 The modified option's new value.
 
 ---
+
+The following example shows how to subscribe to changes of a component option:
```
Comment (AlexJameson) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]: Option or property?

**Review by RomanTsukanov — APPROVED** [review]

File: api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md [hunk 15]
```diff
@@ -15,12 +15,131 @@ Information about the event.
...
-The modified option if it belongs to the first level. Otherwise, the first-level option it is nested into.
+The modified property if it belongs to the first level. Otherwise, the first-level property it is nested into.
 
 ##### field(e.value): any
-The modified option's new value.
+The modified property's new value.
+
+---
+
+The following example shows how to subscribe to changes of a component property:
```
Comment (arminal) [inline api-reference/10 UI Widgets/Component/1 Configuration/onOptionChanged.md]:
```suggestion
The following example shows how to subscribe to component property changes:
```

**Pull request merged**